lib/liquid/block_body.rb

```diff
@@ -13,8 +14,46 @@ def initialize
       @blank = true
     end
 
-    def parse(tokenizer, parse_context)
+    def parse(tokenizer, parse_context, &block)
```

Comment:
Avoiding `&block` and also inlining the two parse methods has a bit of speedup:

```
before:
              parse:     27.170  (± 7.4%) i/s -    270.000  in  10.012500s
             render:     72.857  (± 6.9%) i/s -    728.000  in  10.047589s
     parse & render:     18.784  (±10.6%) i/s -    187.000  in  10.037581s
```

```
after:
              parse:     27.906  (± 7.2%) i/s -    280.000  in  10.072156s
             render:     73.017  (± 9.6%) i/s -    728.000  in  10.085485s
     parse & render:     19.932  (±10.0%) i/s -    199.000  in  10.051317s
```

I don't find it terribly more difficult to read after inlining, but it's also a pretty small difference, within error.

Reply:
Which Ruby version is this benchmark based on? Explicit use of `&block` argument and consequent allocation of Proc was optimized in Ruby 2.6

Reply:
Awesome, that is great to know. My Liquid dev env is on 2.5.3. Will re-test when I come back to this next.

Reply:
:+1: A correction to my comment above:
Proc allocation from `&block` was [optimized in Ruby 2.5](https://bugs.ruby-lang.org/issues/14045)
while Ruby 2.6 brought in [optimizations for `block.call`](https://bugs.ruby-lang.org/issues/14330)